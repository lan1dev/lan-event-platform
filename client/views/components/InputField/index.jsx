import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';
import Text from '../Text';

const propTypes = {
  input: PropTypes.shape({}).isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.shape({}).isRequired
};

const InputField = ({ input, label, type, meta: { touched, error, warning } }) => (
  <div>
    <Input {...input} placeholder={label} type={type} />
    {touched && ((error && <Text error>{error}</Text>) || (warning && <span>{warning}</span>))}
  </div>
);

InputField.propTypes = propTypes;
export default InputField;
