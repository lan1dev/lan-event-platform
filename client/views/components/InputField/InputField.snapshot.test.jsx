import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import InputField from './index';

describe('InputField', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<InputField {...props}>{props.children}</InputField>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      meta: {}
    };
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with error prop', () => {
    it('Should match snapshot', () => {
      props = {
        error: 'an error',
        meta: {
          touched: true
        }
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    describe('when not touched', () => {
      it('Should match snapshot', () => {
        props = {
          warning: 'an warning',
          meta: {
            touched: false
          }
        };
        expect(toJson(Wrapper())).toMatchSnapshot();
      });
    });
  });

  describe('with warning prop', () => {
    it('Should match snapshot', () => {
      props = {
        warning: 'an warning',
        meta: {
          touched: true
        }
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
    describe('when not touched', () => {
      it('Should match snapshot', () => {
        props = {
          warning: 'an warning',
          meta: {
            touched: false
          }
        };
        expect(toJson(Wrapper())).toMatchSnapshot();
      });
    });
  });
});
