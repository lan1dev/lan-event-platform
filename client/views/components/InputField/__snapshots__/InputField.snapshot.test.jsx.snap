// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField with error prop Should match snapshot 1`] = `
<div>
  <styled.input />
</div>
`;

exports[`InputField with error prop when not touched Should match snapshot 1`] = `
<div>
  <styled.input />
</div>
`;

exports[`InputField with warning prop Should match snapshot 1`] = `
<div>
  <styled.input />
</div>
`;

exports[`InputField with warning prop when not touched Should match snapshot 1`] = `
<div>
  <styled.input />
</div>
`;

exports[`InputField without props Should match snapshot 1`] = `
<div>
  <styled.input />
</div>
`;
